### PR TITLE
[Minor] improvement(CI): Remove some unnecessary `gradle build` operation and add conditions to accelerate CI 

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

1. Remove unnecessary builds operation before `gradle compileDistribution`
2. If the backend integration test runs in the embedded mode, there is no need to assemble Gravitino.

### Why are the changes needed?

1. Task `compileDistribution` will compile all modules and copy them to the `distribution` directory except `trio-connector` and `spark-connector`, If the Gravitino server runs in `deploy` mode, this task is a must, if it runs in `embedded` mode, we can skip the task.

image

3. Regardless of the mode the Gravitino server operates in, the integration test will create Trino-related jars by itself when running,  so there is no need to run the build job explicitly in the pipeline. 
 
In conclusion:
- in the `embedded` mode, no need to do anything like `build` and `compileDistribution`
- in the `deploy` mode, task `compileDistribution` is enough. 

@jerryshao 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

N/A.

### CODE REVIEW

File: .github/workflows/backend-integration-test.yml [hunk 1]
```diff
@@ -76,8 +76,8 @@ jobs:
         uses: docker/setup-qemu-action@v2
 
       - name: Package Gravitino
+        if : ${{ matrix.test-mode == 'deploy' }}
         run: |
-          ./gradlew build -x test -PjdkVersion=${{ matrix.java-version }}
```
Comment (jerryshao) [inline .github/workflows/backend-integration-test.yml:80]: Can we remove this step, I remembered that the `compileDistribition` will not build the trino module, this will lead to the test failure, can you please confirm this?

**Comment by jerryshao** [comment]
Also your description is unrelated to your changes.

File: .github/workflows/backend-integration-test.yml (hunk 1 above)
Comment (yuqi1129) [reply .github/workflows/backend-integration-test.yml:80]: image

Task `assembleDistribution` will build the Trino module by the way.

**Comment by yuqi1129** [comment]
> Also your description is unrelated to your changes.

Modified

**Comment by jerryshao** [comment]
> > Also your description is unrelated to your changes.
> 
> Modified

I'm saying the PR description is about `assembleDistribution`, but your changes and the CI script seems doesn't have anything about `assembleDistribution`, so what's the relation here?

**Comment by yuqi1129** [comment]
> > > Also your description is unrelated to your changes.
> 
> > 
> 
> > Modified
> 
> 
> 
> I'm saying the PR description is about `assembleDistribution`, but your changes and the CI script seems doesn't have anything about `assembleDistribution`, so what's the relation here?

Do you means "Remove unnecessary builds operation before gradle assembleDistribution" here?

What I mean here is "REMOVE builds operations" and It is not about "assembleDistribution”

**Comment by yuqi1129** [comment]
@jerryshao 
Module `Integration-test` depends on `trino` task `jar`, so, we can't remove it currently. 

image

**Review by jerryshao — APPROVED** [review]
